pkg/skaffold/cluster/minikube.go

```diff
@@ -0,0 +1,214 @@
+	return false
+}
+
+func (clientImpl) MinikubeExec(arg ...string) (*exec.Cmd, error) {
+	return minikubeExec(arg...)
+}
+
+func minikubeExec(arg ...string) (*exec.Cmd, error) {
+	b, err := minikubeBinaryFunc()
+	if err != nil {
+		return nil, fmt.Errorf("getting minikube executable: %w", err)
+	}
+	return exec.Command(b, arg...), nil
+}
+
+func minikubeBinary() (string, error) {
+	execName := "minikube"
+	if found, _ := util.DetectWSL(); found {
+		execName = "minikube.exe"
+	}
+	filename, err := exec.LookPath(execName)
+	if err != nil {
+		return "", errors.New("unable to find minikube executable. Please add it to PATH environment variable")
+	}
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return "", fmt.Errorf("unable to find minikube executable. File not found %s", filename)
+	}
+	return filename, nil
+}
+
+// matchClusterCertPath checks if the cluster certificate for this context is from inside the minikube directory
+func matchClusterCertPath(kubeContext string) (bool, error) {
+	c, err := getClusterInfo(kubeContext)
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	if c.CertificateAuthority == "" {
+		return false, nil
+	}
+	return util.IsSubPath(minikubePath(), c.CertificateAuthority), nil
+}
+
+// matchProfileAndServerURL checks if kubecontext matches any valid minikube profile
+// and for selected drivers if the k8s server url is same as any of the minikube nodes IPs
+func matchProfileAndServerURL(kubeContext string) (bool, error) {
+	config, err := getRestClientConfigFunc()
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes config: %w", err)
+	}
+	apiServerURL, _, err := rest.DefaultServerURL(config.Host, config.APIPath, schema.GroupVersion{}, false)
+
+	if err != nil {
+		return false, fmt.Errorf("getting kubernetes server url: %w", err)
+	}
+
+	logrus.Tracef("kubernetes server url: %s", apiServerURL)
+
+	ok, err := matchServerURLFor(kubeContext, apiServerURL)
+	if err != nil {
+		return false, fmt.Errorf("checking minikube node url: %w", err)
+	}
+	return ok, nil
+}
+
+func matchServerURLFor(kubeContext string, serverURL *url.URL) (bool, error) {
+	cmd, _ := minikubeExec("profile", "list", "-o", "json")
+	out, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return false, fmt.Errorf("getting minikube profiles: %w", err)
+	}
+
+	var data data
+	if err = json.Unmarshal(out, &data); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data: %w", err)
+	}
+
+	for _, v := range data.Valid {
+		if v.Config.Name != kubeContext {
+			continue
+		}
+
+		if v.Config.Driver != HyperKit && v.Config.Driver != VirtualBox {
+			// Since node IPs don't match server API for other drivers we assume profile name match is enough.
+			// TODO: Revisit once https://github.com/kubernetes/minikube/issues/6642 is fixed
+			return true, nil
+		}
+		for _, n := range v.Config.Nodes {
+			if serverURL.Host == fmt.Sprintf("%s:%d", n.IP, n.Port) {
+				return true, nil
+			}
+		}
```

Comment:
It feels to me like this is inverted:
  - we should check if any of the nodes match the cluster server _regardless of the driver type_.  That way we're insulated from whenever kubernetes/minikube#6642 is fixed.  (At least, my understanding is that the cluster-info's server IP won't match the node IP — my minikube is 127.0.0.1:32780 but the profile lists 172.17.0.2)
  - then see if the minikube profile name matches the kubeContext — this is a heuristic, but unless the user has manually edited their kubeconfig files, it should be true

Reply:
Fixed in https://github.com/GoogleContainerTools/skaffold/pull/4742